### ISSUE TITLE
No longer working!

### ISSUE THREAD

**Issue by mdgleeson** [markathomas/LocationTextField#11]
Hi,
I keep getting the following error. I have not altered the code from this demo. Any help appreciated.

ERROR org.vaadin.addons.locationtextfield.AbstractGeocoderController - Error geocoding query: New York City, NY
org.vaadin.addons.locationtextfield.GeocodingException: A JSONArray text must start with '[' at character 1
        at org.vaadin.addons.locationtextfield.URLConnectionGeocoder.geocode(URLConnectionGeocoder.java:64)
        at org.vaadin.addons.locationtextfield.AbstractGeocoderController.geocode(AbstractGeocoderController.java:55)
        at org.vaadin.addons.locationtextfield.LocationTextField.setText(LocationTextField.java:276)
        at org.vaadin.addons.locationtextfield.LocationTextField.geocode(LocationTextField.java:257)
        at org.vaadin.addons.ltf.demo.DemoUI$2.buttonClick(DemoUI.java:90)

**Comment by markathomas** [comment]
this has nothing to do with the addon.  apparently the return value of the
URL you are using is not an JSON array

**Comment by mdgleeson** [comment]
Thank you for replying.
I use the addon in an application and it has worked perfectly until last week - no code changes made.
To check, I ran your locationtextfield-demo and I am getting the same error. I have made no code changes to your demo or to my own codebase. The URL I am using is whatever URL is in your demo.

Any ideas?

Thank you, much appreciated.

**Comment by markathomas** [comment]
value provided for a valid address is still a JSON array with
OpenStreeMapGeocoder which is what the demo uses.  can you share the
address that is not working?

>

**Comment by mdgleeson** [comment]
"http://nominatim.openstreetmap.org/search?format=json&addressdetails=1&q=" as per demo

If I run your demo as is, I get the same jsonArray error. If you clone/download your demo you should get the same error (I hope!).

**Comment by markathomas** [comment]
that's the base URL.  i'ma asking for the address you're attempting to
geocode.  the return value is correct when i enter a valid address

**Comment by mdgleeson** [comment]
https://nominatim.openstreetmap.org/search?format=json&addressdetails=1&q=New%20York
returns json result in a browser but not through when I use the demo application.

**Comment by markathomas** [comment]
i'll check it out tonight when i have some free time

**Comment by mdgleeson** [comment]
Thank you very much. It will be a great help. Much Appreciated.

**Comment by markathomas** [comment]
it's been fixed in 3.0.1

>

**Comment by mdgleeson** [comment]
Thank you very much for that.
Unfortunately it still returns the same error - both on my application and your demo.
When I cloned your demo (Latest commit 1a06844) I get a java out of bounds error and class loader error. When I updated the pom to 3.0.1 on my app and your old version of demo, I get the JSON array error like before.
Yet the url in a browser still works - its a strange one.

Any ideas much appreciated.
